tests/phpunit/tests/rest-api/rest-widgets-controller.php

```diff
@@ -240,6 +240,46 @@ public function test_get_items_no_permission() {
 		$this->assertErrorResponse( 'rest_cannot_manage_widgets', $response, 401 );
 	}
 
+	public function test_get_items_no_permission_public() {
```

Comment:
I'm confused because the test name says "no permission public", but then it sets up a sidebar with two widgets and the endpoint returns both. I'd add a few test cases where the API returns a limited list of widgets or a 403 error.